src/components/AnalysisName.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Button, Input } from 'semantic-ui-react'
+import * as app from '../store/actions/app'
+
+class AnalysisName extends React.Component {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isEditing: false
+    }
+
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleClick = this.handleClick.bind(this)
+    this.handleCancel = this.handleCancel.bind(this)
+  }
+
+  handleSubmit (event) {
+    let input = this.refs.viewName.inputRef.value
+    if (input === '') {
+      input = 'Untitled Analysis'
```

Comment:
Instead of setting blank values to "Untitled Analysis" let's only show it during `render()` when it wants to display a value, but the value is empty. Internally, we'll still know that the input is empty. (We can assume that this is user intent, and we respect the intent, rather than mutating the value.)